Sources/MusicXML/Complex Types/Metronome.swift

```diff
@@ -75,67 +77,102 @@ extension Metronome {
 extension Metronome.Regular.Relation: Equatable { }
 extension Metronome.Regular.Relation: Codable {
     enum CodingKeys: String, CodingKey {
-        case perMinute
-        case beatUnit
+        case perMinute = "per-minute"
+        case beatUnit = "beat-unit"
+        case beatUnitDot = "beat-unit-dot"
     }
+
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         switch self {
         case let .perMinute(value):
             try container.encode(value, forKey: .perMinute)
-        case let .beatUnit(value):
+        case let .beatUnit(value, dots):
             try container.encode(value, forKey: .beatUnit)
+            try container.encode(dots, forKey: .beatUnitDot)
         }
     }
     public init(from decoder: Decoder) throws {
+        enum DecodingError: Error {
+            case unknownKind
+        }
+
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        do {
+        if container.contains(.perMinute) {
             self = .perMinute(try container.decode(PerMinute.self, forKey: .perMinute))
-        } catch {
-            self = .beatUnit(try container.decode(NoteTypeValue.self, forKey: .beatUnit))
+        } else if container.contains(.beatUnit) {
+            let beatUnit = try container.decode(NoteTypeValue.self, forKey: .beatUnit)
+            let beatUnitDot = try container.decodeIfPresent([Empty].self, forKey: .beatUnitDot)
+            self = .beatUnit(beatUnit, beatUnitDot)
+        } else {
+            throw DecodingError.unknownKind
         }
     }
 }
 
 extension Metronome.Regular: Equatable { }
-extension Metronome.Regular: Codable { }
+extension Metronome.Regular: Codable {
+    enum CodingKeys: String, CodingKey {
+        case beatUnit = "beat-unit"
+        case beatUnitDot = "beat-unit-dot"
+    }
 
-extension Metronome.Complicated: Equatable { }
-extension Metronome.Complicated: Codable { }
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.beatUnit = try container.decode(NoteTypeValue.self, forKey: .beatUnit)
+        self.beatUnitDot = try container.decodeIfPresent([Empty].self, forKey: .beatUnitDot)
+        self.relation = try Relation(from: decoder)
+    }
+}
 
-extension Metronome.Kind: Equatable { }
-extension Metronome.Kind: Codable {
```

Comment:
`Kind` is an artificial struct, so it should not been `Codable` on its own. It should instead be decoded or coded from the `Metronome` `Codable`